lib/bolt/pal.rb

```diff
@@ -228,8 +235,44 @@ def run_task(task_name, targets, params, executor, inventory, &eventblock)
 
     def run_plan(plan_name, params, executor = nil, inventory = nil)
       in_plan_compiler(executor, inventory) do |compiler|
-        r = compiler.call_function('run_plan', plan_name, params)
-        Bolt::PuppetError.convert_puppet_errors(r)
+        result = compiler.call_function('run_plan', plan_name, params)
+        strip_pcore(result, compiler)
+      end
+    end
+
+    # Depth first walk of pcore rich data to convert it into data and bolt ruby objects
```

Comment:
I think limiting Bolt's support to Data makes sense for now. Does that mean we won't display other types nicely, or that they'll cause an error?

Reply:
I think we should error in run_plan to make it clear that you shouldn't write plans that return unexpected types.